core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java

```diff
@@ -184,4 +191,23 @@ protected String getEnvelopName() {
     protected String getBaseTypeName() {
         throw new UnsupportedOperationException();
     }
+    
+    private String normalizeInstrumentationKey(String instrumentationKey){
+    	if (StringUtils.isEmpty(instrumentationKey) || StringUtils.containsOnly(instrumentationKey, ".- ")){
```

Comment:
Is it possible for it to have a missing iKey at this point? or is this an error scenario?

Reply:
outcome will be questionable if key is empty, but it is definitely possible. So thins handling is OK

Reply:
cool. thanks for clarifying